### PR TITLE
Add try/catches to config reading

### PR DESCRIPTION
**Pull request by abaranch**
- Add try/catches to config reading so TelemetryConfiguration.Active does not throw
- Remove couple methods from Platform interface (which needs to be removed all together since now we have only one implementation)

### CODE REVIEW

File: src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 1]
```diff
@@ -29,32 +22,47 @@ public string ReadConfigurationXml()
...
+            try
             {
-                return File.ReadAllText(configFilePath);
+                // Ensure config file actually exists
+                if (File.Exists(configFilePath))
+                {
+                    return File.ReadAllText(configFilePath);
+                }
+            }
+            catch (FileNotFoundException)
```
Comment (dnduffy) [inline src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs]: How can there be a FileNotFoundException if you just checked File.Exists?

File: src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 2]
```diff
@@ -29,32 +22,47 @@ public string ReadConfigurationXml()
...
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (SecurityException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
             }
```
Comment (dnduffy) [inline src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs]: What about PathTooLongException and NotSupportedException?

File: src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs [hunk 3]
```diff
@@ -29,32 +22,47 @@ public string ReadConfigurationXml()
...
+                // For cases when file was deleted/modified while reading
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (UnauthorizedAccessException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
+            }
+            catch (SecurityException)
+            {
+                CoreEventSource.Log.ApplicationInsightsConfigNotFoundWarning(configFilePath);
```
Comment (dnduffy) [inline src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs]: What about PathTooLongException and NotSupportedException?

File: src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 1 above)
Comment (abaranch) [inline src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs]: Yes, you got it right

---
In reply to: [77934014](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/304#discussion_r77934014) [](ancestors = 77934014)

File: src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs (hunk 3 above)
Comment (abaranch) [inline src/Core/Managed/Net40/Extensibility/Implementation/Platform/PlatformImplementation.cs]: PathTooLong is IOException. NotSupportedException is thrown when path is in incorrect format - we create path on line 23; it is not possible to have incorrect format.

---
In reply to: [77935073](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/304#discussion_r77935073) [](ancestors = 77935073)

**Comment by dnduffy** [comment]
:+1:

**Pull request merged**